src/script.cpp

```diff
@@ -1605,10 +1616,27 @@ bool IsMine(const CKeyStore &keystore, const CScript& scriptPubKey)
         // them) enable spend-out-from-under-you attacks, especially
         // in shared-wallet situations.
         vector<valtype> keys(vSolutions.begin()+1, vSolutions.begin()+vSolutions.size()-1);
-        return HaveKeys(keys, keystore) == keys.size();
+        if (HaveKeys(keys, keystore) == keys.size()) {
+            ret = std::max(ret, IsMineResult::SPENDABLE);
+        }
+        break;
     }
     }
-    return false;
+    // TODO: Watch-only addresses
```

Comment:
I think we don't store watch-only addresses at all yet, right?

Reply:
Correct, but we should eventually support it...